src/bin/cargo-flamegraph.rs

```diff
@@ -270,32 +271,25 @@ fn find_unique_bin_target() -> anyhow::Result<BinaryTarget> {
         .flat_map(|p| {
             let Package { targets, name, .. } = p;
             targets.into_iter().filter_map(move |t| {
-                t.kind.iter().any(|s| s == "bin").then(|| BinaryTarget {
-                    package: name.clone(),
-                    target: t.name,
-                })
+                t.kind
+                    .iter()
+                    .any(|s| kind.contains(&s.as_str()))
+                    .then(|| BinaryTarget {
+                        package: name.clone(),
+                        target: t.name,
+                    })
             })
         })
         .collect();
 
-    match bin_targets.as_slice() {
+    match targets.as_slice() {
         [_] => {
-            let target = bin_targets.remove(0);
-            eprintln!(
-                "automatically selected {} as it is the only binary target",
-                target
-            );
+            let target = targets.remove(0);
+            eprintln!("automatically selected {} as it is the only valid target", target);
             Ok(target)
         }
-        [] => Err(anyhow!(
-            "crate has no binary targets: try passing `--example <example>` \
-                or similar to choose a binary"
-        )),
-        _ => Err(anyhow!(
-            "several possible targets found: {:?}, please pass `--bin <binary>` or \
-                `--example <example>` to cargo flamegraph to choose one of them",
-            bin_targets
-        )),
+        [] => Err(anyhow!("crate has no automatically selectable target: try passing `--example <example>` or similar to choose a binary")),
```

Comment:
I'd prefer that you don't change the formatting in the same commit as making code changes, as it makes it harder to review what you actually changed. In this case, I also prefer the previous formatting (that is, break long literals apart so they don't pass 100 columns).

Reply:
ok, I reverted the formatting and only adjusted the text.

Reply:
Thanks!